### PR TITLE
[Alpha pipeline] Add segmentation component

### PR DESCRIPTION
**Pull request by NielsRogge**
This PR adds the final component of the ControlNet pipeline, namely the segmentation one.

### CODE REVIEW

File: examples/pipelines/controlnet-interior-design/run_locally.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
+export METADATA='{"run_id":"test_niels", "base_path":"/Users/nielsrogge/Documents/fondant_artifacts"}'
```
Comment (NielsRogge) [inline examples/pipelines/controlnet-interior-design/run_locally.sh:1]: Will remove this script when PR gets approved

File: examples/pipelines/controlnet-interior-design/components/segment_images/src/main.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        Returns:
+            Dask dataframe
+        """
+        device = "cuda" if torch.cuda.is_available() else "cpu"
+        logger.info("Device:", device)
+
+        processor = AutoImageProcessor.from_pretrained(model_id)
+        model = AutoModelForSemanticSegmentation.from_pretrained(model_id)
+
+        print("Length of the input dataframe:", len(dataframe))
```
Comment (PhilippeMoussalli) [inline examples/pipelines/controlnet-interior-design/components/segment_images/src/main.py:108]: Don't forget to remove those

File: examples/pipelines/controlnet-interior-design/components/segment_images/src/palette.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
+import numpy as np
```
Comment (PhilippeMoussalli) [inline examples/pipelines/controlnet-interior-design/components/segment_images/src/palette.py:1]: Could you add a module docstring with a brief description and maybe a reference to where this was taken from?

File: examples/pipelines/controlnet-interior-design/components/segment_images/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,3 @@
+git+https://github.com/ml6team/fondant.git@main
```
Comment (PhilippeMoussalli) [inline examples/pipelines/controlnet-interior-design/components/segment_images/requirements.txt:1]: should numpy be here? or is it installed with transformers?

**Review by PhilippeMoussalli — COMMENTED** [review]
Thanks @NielsRogge :) Seems like the development of the components is going fast now which is nice to see. 

Left a few comments. 

I feel like a lot of the dask inference stuff can be packaged into a common util at some point to make it easier to develop (not part of this PR). Wdyt?

File: examples/pipelines/controlnet-interior-design/components/segment_images/requirements.txt (hunk 4 above)
Comment (NielsRogge) [reply examples/pipelines/controlnet-interior-design/components/segment_images/requirements.txt:1]: Should be fine when Transformers is installed

File: examples/pipelines/controlnet-interior-design/build_images.sh [hunk 5]
```diff
@@ -50,6 +50,7 @@ for dir in $component_dir/*/; do
      --build-arg BUILD_TIMESTAMP=$(date '+%F_%H:%M:%S') \
      --label org.opencontainers.image.source=https://github.com/${namespace}/${repo} \
      --platform=linux/arm64 \
+     --no-cache \
```
Comment (RobbeSneyders) [inline examples/pipelines/controlnet-interior-design/build_images.sh:53]: Do we need to commit this?

**Review by RobbeSneyders — APPROVED** [review]
Thanks @NielsRogge!

Don't forget to remove the script.

**Pull request merged**